waku/v2/node/waku_node.nim

```diff
@@ -879,17 +879,12 @@ proc keepaliveLoop(node: WakuNode, keepalive: chronos.Duration) {.async.} =
                                 .filterIt(it.connectedness == Connected)
                                 .mapIt(it.toRemotePeerInfo())
 
-    # Attempt to retrieve and ping the active outgoing connection for each peer
     for peer in peers:
-      let connOpt = await node.peerManager.dialPeer(peer, PingCodec)
-
-      if connOpt.isNone():
-        # TODO: more sophisticated error handling here
-        debug "failed to connect to remote peer", peer=peer
+      try:
+        let conn = await node.switch.dial(peer.peerId, peer.addrs, PingCodec)
+        let pingDelay = await node.libp2pPing.ping(conn)
+      except CatchableError as exc:
```

Comment:
The main change here is to use `switch.dial` instead of `peerManager.dialPeer` for the ping. Main reason is that `dialPeer` updates metrics on ok/nok connections, failed attempts, last time failed, Can/Cannot be connected etc.

And since here we are just pinging this peer, I don't think using that function makes sense. For example, we will be updating the metrics with ok connections, every time we ping a peer.

And actually this should be more like `getConnection` because we are not dialing/connecting to any peer, but getting an already existing connection and sending a ping.

I agree with explicir error handling, but here I had to add the `try` `except` because `switch.dial` doesn't return `Result[xx]`.